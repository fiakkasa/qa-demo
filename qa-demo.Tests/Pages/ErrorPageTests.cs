using System.Net;

namespace qa_demo.Pages.Tests;

public class ErrorPageTests(PagesFixture fixture) : IClassFixture<PagesFixture>
{
    [Fact]
    public async Task ErrorPage_Should_Render()
    {
        var result = await fixture.Client.GetAsync("/error");

        result.StatusCode.Should().Be(HttpStatusCode.OK);

        var (html, htmlDoc) = await result.GetHtmlAssets();

        htmlDoc.DocumentNode.SelectSingleNode("//html/head/title").InnerText.Should().Be("Error - QA Demo");
        
        var dangerText = htmlDoc.QuerySelectorAll(".text-danger");
        dangerText.Should().HaveCount(2);
        dangerText[0].InnerText.Should().Be("Error.");
        dangerText[0].Name.Should().Be("h1");
        dangerText[1].InnerText.Should().Be("An error occurred while processing your request.");
        dangerText[1].Name.Should().Be("h2");

        html.MatchSnapshot();
    }
}