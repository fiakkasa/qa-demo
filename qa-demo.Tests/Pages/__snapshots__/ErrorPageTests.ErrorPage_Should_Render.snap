<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Error - QA Demo</title>
    <link rel="stylesheet" href="/lib/bootstrap/dist/css/bootstrap.min.css" />
    <link rel="stylesheet" href="/css/site.css?v=pAGv4ietcJNk_EwsQZ5BN9-K4MuNYS2a9wl4Jw-q9D0" />
    <link rel="stylesheet" href="/qa_demo.styles.css" />
</head>
<body>
    <header b-16aypfayd1>
        <nav b-16aypfayd1 class="navbar navbar-expand-sm navbar-toggleable-sm navbar-light bg-white border-bottom box-shadow mb-3">
            <div b-16aypfayd1 class="container">
                <a class="navbar-brand" href="/">QA Demo</a>
                <button b-16aypfayd1 class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target=".navbar-collapse" aria-controls="navbarSupportedContent"
                        aria-expanded="false" aria-label="Toggle navigation">
                    <span b-16aypfayd1 class="navbar-toggler-icon"></span>
                </button>
                <div b-16aypfayd1 class="navbar-collapse collapse d-sm-inline-flex justify-content-between">
                    <ul b-16aypfayd1 class="navbar-nav flex-grow-1">
                        <li b-16aypfayd1 class="nav-item">
                            <a class="nav-link text-dark" href="/">Home</a>
                        </li>
                        <li b-16aypfayd1 class="nav-item">
                            <a class="nav-link text-dark" href="/Privacy">Privacy</a>
                        </li>
                    </ul>
                </div>
            </div>
        </nav>
    </header>
    <div b-16aypfayd1 class="container">
        <main b-16aypfayd1 role="main" class="pb-3">
            
<h1 class="text-danger">Error.</h1>
<h2 class="text-danger">An error occurred while processing your request.</h2>

        </main>
    </div>

    <footer b-16aypfayd1 class="border-top footer text-muted">
        <div b-16aypfayd1 class="container">
            &copy; 2024 - QA Demo - <a href="/Privacy">Privacy</a>
        </div>
    </footer>

    <script src="/lib/jquery/dist/jquery.min.js"></script>
    <script src="/lib/bootstrap/dist/js/bootstrap.bundle.min.js"></script>
    <script src="/js/site.js?v=hRQyftXiu1lLX2P9Ly9xa4gHJgLeR1uGN5qegUobtGo"></script>

    
</body>
</html>
